### DISCUSSION TITLE
Add an `icon` field for chains

### DISCUSSION THREAD

**Discussion by akasuv** [wagmi-dev/references#414] [Ideas]
Most of the time when we need to use a chain's info, we also need the icon of the chain. It would be great if we add a field called `icon` for each chain to display the official icon, and the value should be SVG to avoid the resource hosting problem.

**Comment by nero-sensei** [comment]
How are icons fetched as of now ?

**Reply by akasuv** [reply]
We save them in local folders and map the resource by chain id. I think it would be better if the package embedded a default icon or at least provided an image URL config option.

**Reply by glitch-txs** [reply]
Wouldn't this make the package too heavy though? As there are more and more chains each day

**Reply by akasuv** [reply]
Yes, url config is more reasonable, similar to the RPC provider config.

**Reply by nero-sensei** [reply]
a url config would really helpful.
